File: server/user-service/src/main/java/com/mshop/userservice/controller/dto/UserDto.java
package com.mshop.userservice.controller.dto;

import com.mshop.userservice.repository.entitity.Gender;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private String userId;

    private String fullName;

    private String emailOrPhoneNumber;

    private LocalDate dob;

    private String bio;

    private String avatar;

    private String cover;

    private AddressDto address;

    private Gender gender;
}

File: server/auth-service/src/main/kotlin/com/mshop/authservice/dto/UserDto.kt
package com.mshop.authservice.dto

import com.mshop.authservice.model.Gender
import java.time.LocalDate

data class UserDto(
    val userId: String,
    val fullName: String,
    val emailOrPhoneNumber: String,
    val dob: LocalDate = LocalDate.now(),
    val bio: String = "New user",
    val avatar: String = "user_default.png",
    val cover: String = "user_default.png",
    val address: com.mshop.authservice.dto.AddressDto = com.mshop.authservice.dto.AddressDto(),
    val gender: Gender = Gender.OTHER,
)